jakecache-sw.js

```diff
@@ -116,7 +116,7 @@ class JakeCacheManifest {
 self.addEventListener('message', function (event) {
   switch (event.data.command) {
     case 'update':
-      update.call(this, event.data.pathname, event.data.options)
+      event.waitUntil(update.call(this, event.data.pathname, event.data.options))
```

Comment:
update.call will call it immediately when that line is reached. I think you want to wrap in into a function like

```
event.waitUntil(() => update.call(this, event.data.pathname, event.data.options)))
```

Reply:
[waitUntil](https://developer.mozilla.org/en-US/docs/Web/API/ExtendableEvent/waitUntil#parameters) expects a promise, not a function

Reply:
Ah, I see. But then why is this needed? update is not async (or returning a promise) so that work (the call) should be completed before it being terminated.

Reply:
Ah I was looking at the wrong update :)